Flickr/Flickr/GalleryViewingCollectionViewController.swift

```diff
@@ -8,12 +8,64 @@
 
 import UIKit
 
-class GalleryViewingCollectionViewController: UIViewController {
+class GalleryViewingCollectionViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    
+    var photoGalleryData = [Photo]()
+    var main = ImageCollectionViewController()
+    var indexCell: IndexPath?
     
-    var photoGalleryData: [Photo]?
-
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        self.view.addSubview(collectionView)
+        collectionView.alpha = 0
+    }
+    @IBOutlet weak var collectionView: UICollectionView!
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        guard let indexPath = indexCell else {
+            return
+        }
+        collectionView?.scrollToItem(at: indexPath, at: .left, animated: false)
+        collectionView.alpha = 1
+    }
+    
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        numberOfItemsInSection section: Int) -> Int {
+        print("\(photoGalleryData.count)")
+        return photoGalleryData.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView,
+                        cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cellPopular = collectionView.dequeueReusableCell(withReuseIdentifier: "Gallery Image Cell",
+                                                             for: indexPath)
+        if let imageCell = cellPopular as? GalleryViewingCollectionViewCell {
```

Comment:
in this case you can use guard instead of "if". Also GalleryViewingCollectionViewCell -> ImageDetailCollectionViewCell